Yafc/Workspace/ProductionTable/ModuleFillerParametersScreen.cs

```diff
@@ -50,31 +90,90 @@ public class ModuleFillerParametersScreen : PseudoScreen {
-                });
+            if (defaultBeacon is null || defaultBeaconModule is null) {
+                gui.BuildText("Your mods contain no beacons, or no modules that can be put into beacons.");
             }
+            else {
+                if (gui.BuildFactorioObjectButtonWithText(modules.beacon) == Click.Left) {
+                    SelectSingleObjectPanel.SelectWithNone(Database.allBeacons, "Select beacon", select => {
+                        _ = modules.RecordUndo();
+                        modules.beacon = select;
+                        if (modules.beaconModule != null && (modules.beacon == null || !modules.beacon.CanAcceptModule(modules.beaconModule.moduleSpecification))) {
+                            modules.beaconModule = null;
+                        }
 
-            if (gui.BuildFactorioObjectButtonWithText(modules.beaconModule) == Click.Left) {
-                SelectSingleObjectPanel.SelectWithNone(Database.allModules.Where(x => modules.beacon?.CanAcceptModule(x.moduleSpecification) ?? false), "Select module for beacon", select => { modules.RecordUndo().beaconModule = select; });
-            }
+                        gui.Rebuild();
+                    });
+                }
 
-            using (gui.EnterRow()) {
-                gui.BuildText("Beacons per building: ");
-                if (gui.BuildTextInput(modules.beaconsPerBuilding.ToString(), out string newText, null, Icon.None, true, new Padding(0.5f, 0f)) &&
-                    int.TryParse(newText, out int newAmount) && newAmount > 0) {
-                    modules.RecordUndo().beaconsPerBuilding = newAmount;
+                if (gui.BuildFactorioObjectButtonWithText(modules.beaconModule) == Click.Left) {
+                    SelectSingleObjectPanel.SelectWithNone(Database.allModules.Where(x => modules.beacon?.CanAcceptModule(x.moduleSpecification) ?? false), "Select module for beacon", select => { modules.RecordUndo().beaconModule = select; });
+                }
+
+                using (gui.EnterRow()) {
+                    gui.BuildText("Beacons per building: ");
+                    if (gui.BuildTextInput(modules.beaconsPerBuilding.ToString(), out string newText, null, Icon.None, true, new Padding(0.5f, 0f)) &&
+                        int.TryParse(newText, out int newAmount) && newAmount > 0) {
+                        modules.RecordUndo().beaconsPerBuilding = newAmount;
+                    }
+                }
+                gui.BuildText("Please note that beacons themselves are not part of the calculation", wrap: true);
+
+                gui.AllocateSpacing();
+                gui.BuildText("Override beacons:", Font.subheader);
+                if (modules.overrideCrafterBeacons.Count > 0) {
+                    gui.BuildText("  Click to change beacon, right-click to change module", topOffset: -0.5f);
+                    gui.BuildText("  Select the 'none' item in either prompt to remove the override.", topOffset: -0.5f);
```

Comment:
Instead of prefixing the string with spaces, wouldn't it work to add a left offset instead? (assuming that is what the spaces are for)